File: src/lib/api/system.ts
import { invoke } from '@tauri-apps/api/core'

/**
 * 重启应用
 */
export async function restartApp(): Promise<void> {
  await invoke('restart')
}

/**
 * 获取当前运行平台类型 (windows, macos, android, ios, linux)
 */
export async function getPlatform(): Promise<string> {
  return await invoke<string>('get_platform')
}

/**
 * 示例 Greet 命令
 * @param name 姓名
 */
export async function greet(name: string): Promise<string> {
  return await invoke<string>('greet', { name })
}

File: src-tauri/src/commands/system.rs
use log::info;
use tauri::AppHandle;

#[tauri::command]
pub fn restart(app: AppHandle) {
    info!("正在重启应用...");
    app.restart();
}

#[tauri::command]
pub fn get_platform() -> String {
    // 返回当前操作系统平台：windows, macos, linux, android, ios
    if cfg!(target_os = "windows") {
        "windows".to_string()
    } else if cfg!(target_os = "macos") {
        "macos".to_string()
    } else if cfg!(target_os = "android") {
        "android".to_string()
    } else if cfg!(target_os = "ios") {
        "ios".to_string()
    } else if cfg!(target_os = "linux") {
        "linux".to_string()
    } else {
        "unknown".to_string()
    }
}
